thorclient/thorclient.go

```diff
@@ -0,0 +1,274 @@
+	options := applyOptions(opts)
+	return c.httpConn.GetTransactionReceipt(id, options.revision)
+}
+
+// SendTransaction sends a signed transaction to the blockchain.
+func (c *Client) SendTransaction(tx *tx.Transaction) (*transactions.SendTxResult, error) {
+	rlpTx, err := rlp.EncodeToBytes(tx)
+	if err != nil {
+		return nil, fmt.Errorf("unable to encode transaction - %w", err)
+	}
+
+	return c.SendRawTransaction(rlpTx)
+}
+
+// SendRawTransaction sends a raw RLP-encoded transaction to the blockchain.
+func (c *Client) SendRawTransaction(rlpTx []byte) (*transactions.SendTxResult, error) {
+	return c.httpConn.SendTransaction(&transactions.RawTx{Raw: hexutil.Encode(rlpTx)})
+}
+
+// Block retrieves a block by its revision.
+func (c *Client) Block(revision string) (blocks *blocks.JSONCollapsedBlock, err error) {
+	return c.httpConn.GetBlock(revision)
+}
+
+// ExpandedBlock retrieves an expanded block by its revision.
+func (c *Client) ExpandedBlock(revision string) (blocks *blocks.JSONExpandedBlock, err error) {
+	return c.httpConn.GetExpandedBlock(revision)
+}
+
+// FilterEvents filters events based on the provided filter request.
+func (c *Client) FilterEvents(req *events.EventFilter) ([]events.FilteredEvent, error) {
+	return c.httpConn.FilterEvents(req)
+}
+
+// FilterTransfers filters transfers based on the provided filter request.
+func (c *Client) FilterTransfers(req *events.EventFilter) ([]*transfers.FilteredTransfer, error) {
+	return c.httpConn.FilterTransfers(req)
+}
+
+// Peers retrieves the list of connected peers.
+func (c *Client) Peers() ([]*node.PeerStats, error) {
+	return c.httpConn.GetPeers()
+}
+
+// ChainTag retrieves the chain tag from the genesis block.
+func (c *Client) ChainTag() (byte, error) {
+	genesisBlock, err := c.Block("0")
+	if err != nil {
+		return 0, err
+	}
+	return genesisBlock.ID[31], nil
+}
+
+// SubscribeBlocks subscribes to block updates over WebSocket.
+func (c *Client) SubscribeBlocks() (*common.Subscription[*blocks.JSONCollapsedBlock], error) {
+	if c.wsConn == nil {
+		return nil, fmt.Errorf("not a websocket typed client")
+	}
+	return c.wsConn.SubscribeBlocks("")
+}
+
+// SubscribeEvents subscribes to event updates over WebSocket.
+func (c *Client) SubscribeEvents() (*common.Subscription[*subscriptions.EventMessage], error) {
+	if c.wsConn == nil {
+		return nil, fmt.Errorf("not a websocket typed client")
+	}
+	return c.wsConn.SubscribeEvents("")
+}
+
+// SubscribeTransfers subscribes to transfer updates over WebSocket.
+func (c *Client) SubscribeTransfers() (*common.Subscription[*subscriptions.TransferMessage], error) {
+	if c.wsConn == nil {
+		return nil, fmt.Errorf("not a websocket typed client")
+	}
+	return c.wsConn.SubscribeTransfers("")
+}
+
+// SubscribeBeats2 subscribes to Beat2 message updates over WebSocket.
+func (c *Client) SubscribeBeats2() (*common.Subscription[*subscriptions.Beat2Message], error) {
```

Comment:
Query `pos` is required for all subscriptions.

Reply:
Good spot. Fixed 👍